src/Databases/SQLite/DatabaseSQLite.cpp

```diff
@@ -30,15 +32,15 @@ DatabaseSQLite::DatabaseSQLite(
     , database_engine_define(database_engine_define_->clone())
     , log(&Poco::Logger::get("DatabaseSQLite"))
 {
+    auto db_path = SQLiteDatabaseValidatePath(database_path_, context_->getUserFilesPath());
+
     sqlite3 * tmp_sqlite_db = nullptr;
-    int status = sqlite3_open(database_path_.c_str(), &tmp_sqlite_db);
+    int status = sqlite3_open(db_path.c_str(), &tmp_sqlite_db);
```

Comment:
Create separate helper function that will return std::shared_ptr<sqlite3> something like connectionOpenSQLite3 and reuse it in StorageSQLite.cpp, TableFunctionSQLite.cpp.

Reply:
I think it is redundant, because all methods in sqlite are like this one - require to check statuses and etc - and they are also used similarly in database engine and table engine, so then creating a helper function only for opening sqlite database and not for other methods - does not seem consistent.

Reply:
@kssenii this is RAII style if we use sqlite3_open we need to put it in some sort of wrapper immediately, other function does not require to have wrapper because they do not acquire resources.